app/controllers/UploadProxyController.scala

```diff
@@ -148,19 +227,26 @@ private object UploadProxyController {
     }
 
     private def toErrorFields(key: String, xmlResponseBody: String): Seq[(String, String)] =
-      (KeyName -> key) +: xmlErrorFields(xmlResponseBody)
+      toKeyField(key) +: xmlErrorFields(xmlResponseBody)
+
+    private def toKeyField(key: String): (String, String) =
```

Comment:
splitting this out to it's own function seems like a unnecessary level of indirection